corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -119,14 +121,26 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
                 }
                 else {
                     // Set lookup table option by label
-                    var option = domElement.find("option").filter(function (_) {
-                        return $(this).text().trim() === value;
-                    });
-                    if (option.length === 1) {
-                        domElement.val(String(option.index() - 1)).trigger("change");
-                    } else {
-                        domElement.val("-1").trigger('change');
+                    var matchingOption = function (el) {
+                        return el.find("option").filter(function (_) {
+                            return $(this).text().trim() === value;
+                        });
                     }
+                    domElement.each(function () {
```

Comment:
I don't follow how this is working. `domElement` should only be a single element, in this case the `select`. So why iterate over it?

Below where you're calling `matchingOption` won't that still only ever return a single matched option? And won't `value` be a list or something?

Reply:
> I don't follow how this is working. domElement should only be a single element, in this case the select. So why iterate over it?

`domElement` is a jquery selector and would map to multiple matching elements in case when there are multiple select/multiselect search fields. Iterating would apply this logic to each of those fields

> Below where you're calling matchingOption won't that still only ever return a single matched option? And won't value be a list or something?

`value` will be a string returned from geocoder utils. `matchingOption` is called for each field, so it will return different option for different fields.

Reply:
OK, now I'm really confused, how does multi-select interact with geocoder fields? I see now what the code was doing before but I don't understand why it has to change for multi-select. Surely you can't have a multi-select geocoder field right?

Reply:
Calling the vars `domElements` and `matchingOptions` would help clarify that those can be one or more items.

Reply:
Just saw Simon's last comment and looked at the broader context of this code, and now I'm confused as well, I also don't see why a geocoder receiver would ever have multiple values. The line with `concat` makes me think if you did set up a multiselect receiver, say for state, (I don't know why you would, but ignore that for a second), if you selected an address from the geocoder and then changed you mind and selected a different address, the multiselect field would end up with both values, even though the user's intention was to overwrite the previous address with the new one.

Reply:
I designed this intentionally to keep the geocoder receive consistent with single selects. My idea of the usecase was if user wants to lookup locations from multiple cities/streets. Each time geocoder is updated, an extra field will get added to multiselect. The other options are
1. to override all the extra selections and keep just the new geocoder receive field.
2. Not support geocoder with multiselect at all.

2 was easy but I wanted to add the support. 1 would mean breaking the users selections whenever geocoder field is updated. It's easier to clear away extra selections than select those again that were removed by 1.

> Calling the vars domElements and matchingOptions would help clarify that those can be one or more items.

`matchingOption` is a function that takes a single element, but agreed about `domElements`.

Reply:
Actually, I take some of this back. Specificlly
> domElement is a jquery selector and would map to multiple matching elements in case when there are multiple select/multiselect search fields. Iterating would apply this logic to each of those fields

domElement is a jquery Object and I was trying to get the Javascript DOM object, which I had done using `domElement.each`. But this was not the neat way to do that. I have now updated the code to remove that and it's much simpler.

Reply:
I'm still a bit skeptical that there's a use case for multiple address selection, but defer to you since you've thought more about this feature. Agreed that the newer version of the code is easier to follow.